File: feedback_pipeline/generations/cpp/files_with_feedback/question-81-pass-1.cpp
#include <iostream>
using namespace std;
int countWays(int n) {
    int dp[n+1];
    memset(dp, 0, sizeof(dp));
    dp[0] = 1;
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= i-1; j++) {
            if (j + dp[i-j] <= i) {
                dp[i]++;
            }
        }
    }
    return dp[n];
}
int main() {
    int n;
    cout << "Enter a positive integer: ";
    cin >> n;
    cout << "The count of ways to write " << n << " as a sum of three numbers is: " << countWays(n) << endl;
    return 0;
}

File: feedback_pipeline/generations/python/files_with_feedback/question-81-pass-1.py
def countWays(N):
    if N <= 0:
        return 0
    if N == 1:
        return 1
    if N == 2:
        return 2
    if N == 3:
        return 3
    return countWays(N-1) + countWays(N-2) + countWays(N-3)
